pkg/kubelet/kubelet_node_status.go

```diff
@@ -542,24 +544,25 @@ func (kl *Kubelet) updateNodeStatus(ctx context.Context) error {
 
 // tryUpdateNodeStatus tries to update node status to master if there is any
 // change or enough time passed from the last sync.
-func (kl *Kubelet) tryUpdateNodeStatus(ctx context.Context, tryNumber int) error {
+func (kl *Kubelet) tryUpdateNodeStatus(ctx context.Context, useCache bool) error {
 	// In large clusters, GET and PUT operations on Node objects coming
 	// from here are the majority of load on apiserver and etcd.
-	// To reduce the load on control-plane, we are serving GET operations from
-	// local lister (the data might be slightly delayed but it doesn't
-	// seem to cause more conflict - the delays are pretty small).
-	// If it result in a conflict, all retries are served directly from etcd.
-	var originalNode *v1.Node
-	var err error
-
-	if tryNumber == 0 {
-		originalNode, err = kl.nodeLister.Get(string(kl.nodeName))
-	} else {
-		opts := metav1.GetOptions{}
-		originalNode, err = kl.heartbeatClient.CoreV1().Nodes().Get(ctx, string(kl.nodeName), opts)
-	}
+	// To reduce the load on both apiserver and etcd, we are getting Node
+	// object from local cache, which is synchronized via informer/watch (the
+	// data might be slightly delayed but it doesn't seem to cause more
+	// conflict - the delays are pretty small).
+	// If it result in a conflict, the following retry will be served directly
+	// from etcd.
+	originalNode, err := kl.GetNode()
```

Comment:
if you don't use the cache why do you need to get the node from the informer if you are getting it directly later ?

Reply:
Just for code simplicity, otherwise I need to declare an empty Node object like:

```
node := v1.Node{}
if useCache {
    node = kl.GetNode()
} else {
    node = kl.client.CoreV1().Nodes().Get()
}
```

I assumed getting the node from the informer is very cheap here.

Reply:
code simplicity as in removing lines? this will be compiled so not much benefit on having less lines, it also seems harder to ready and to understand